pkg/products/monitoring/reconciler.go

```diff
@@ -457,19 +459,51 @@ func (r *Reconciler) reconcileScrapeConfigs(ctx context.Context, serverClient k8
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
-func (r *Reconciler) reconcileTemplates(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
-	// Iterate over template_list
-	for _, template := range r.Config.GetTemplateList() {
-		// create it
-		_, err := r.createResource(ctx, template, serverClient)
+func (r *Reconciler) reconcileDashboards(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+
+	for _, dashboard := range r.Config.GetDashboards() {
+		err := r.reconcileGrafanaDashboards(ctx, serverClient, dashboard)
 		if err != nil {
-			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create/update monitoring template %s: %w", template, err)
+			logrus.Errorf("Error reconciling dashboards: %v", err)
+			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create/update grafana dashboard %s: %w", dashboard, err)
 		}
-		r.Logger.Infof("Reconciling the monitoring template %s was successful", template)
+		r.Logger.Infof("Reconciling the grafana dashboard  %s was successful", dashboard)
 	}
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+func (r *Reconciler) reconcileGrafanaDashboards(ctx context.Context, serverClient k8sclient.Client, dashBoard string) (err error) {
+
+	grafanaDB := &grafanav1alpha1.GrafanaDashboard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dashBoard,
+			Namespace: r.Config.GetOperatorNamespace(),
+			Labels: map[string]string{
+				"monitoring-key": r.Config.GetLabelSelector(),
```

Comment:
I think the setting of this label should be moved to the mutate function that's passed to the `controllerutil.CreateOrUpdate` (the final func parameter there), since otherwise there's a chance that it won't be re-added if it's removed.

Reply:
Agreed. Fixed.